flink-runtime/src/main/java/org/apache/flink/runtime/state/TaskLocalStateStoreImpl.java

```diff
@@ -186,20 +187,27 @@ private void persistLocalStateMetadata(long checkpointId, TaskStateSnapshot loca
 
     @VisibleForTesting
     File getTaskStateSnapshotFile(long checkpointId) {
-        final File checkpointDirectory =
-                localRecoveryConfig
-                        .getLocalStateDirectoryProvider()
-                        .orElseThrow(
-                                () -> new IllegalStateException("Local recovery must be enabled."))
-                        .subtaskSpecificCheckpointDirectory(checkpointId);
+        return new File(getCheckpointDirectory(checkpointId), TASK_STATE_SNAPSHOT_FILENAME);
+    }
 
+    private File createTaskStateSnapshotFile(long checkpointId) {
```

Comment:
The method doesn't actually create the file (rather the folder).

I think the following set of methods would be more clear:
- `getTaskStateSnapshotFile(String path, long checkpointId)` // add filename suffix
- `getCheckpointDirectory(checkpointId)` // as is
- `createFolderOrFail(String path)` // mkdirs

WDYT?

Reply:
Nice suggestion. I have changed it into(the parameters are a little bit different ):
- `getTaskStateSnapshotFile(long checkpointId)` // add filename suffix
- `getCheckpointDirectory(checkpointId)` // as is
- `createFolderOrFail(File checkpointDirectory)` // mkdirs